lib/grpc_or_grpcweb.dart

```diff
@@ -13,8 +13,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import 'src/client/grpc_or_grpcweb_channel_grpc.dart'
-    if (dart.library.html) 'src/client/grpc_or_grpcweb_channel_web.dart';
```

Comment:
You can avoid the swap and use dart.library.js_interop in place of dart.library.html. There's some documentation here: https://dart.dev/interop/js-interop/package-web that has an example of that.